compiler/il/OMRNode.cpp

```diff
@@ -5173,6 +5173,27 @@ OMR::Node::printIsNonZero()
    return self()->isNonZero() ? "X!=0 " : "";
    }
 
+bool                   OMR::Node::profileInfoCannotBeTrusted()    { return _flags.testAny(profileInfoIsUntrustworthy); }
+void                   OMR::Node::setProfileInfoCannotBeTrusted() {  _flags.set(profileInfoIsUntrustworthy); }
+const char *           OMR::Node::printIsProfileInfoCannotBeTrusted() { return self()->profileInfoCannotBeTrusted() ? "profileInfoIsUntrustworthy " : ""; }
+ 
+void OMR::Node::setProfileInfoCannotBeTrusted(bool v)
+       {
+       TR::Compilation * c = TR::comp();
+       if (performNodeTransformation2(c,"O^O NODE FLAGS: Setting profileInfoIsUntrustworthy flag on node %p to %d\n", this, v))
+          _flags.set(profileInfoIsUntrustworthy, v);
+       }
+ 
+bool OMR::Node::canCheckIfProfileInfoCannotBeTrusted()
```

Comment:
This seems like something that should return false by default and an extensible class should implement the below logic. This seems like "J9-ish" type logic in the base `OMR::Node` class. Do others agree?